src/main/java/org/terasology/minion/move/FindPathToNode.java

```diff
@@ -61,26 +65,19 @@ public void construct(final Actor actor) {
             moveComponent.path = Path.INVALID;
             return;
         }
-        SettableFuture<List<Path>> pathFuture = pathfinderSystem.requestPath(
-                actor.getEntity(), currentBlock.getBlockPosition(),
-                Arrays.asList(workTarget.getBlockPosition()));
+        pathfinderSystem.requestPath(
+                        actor.getEntity(), currentBlock.getBlockPosition(),
+                        Collections.singletonList(workTarget.getBlockPosition()))
+                .publishOn(GameScheduler.gameMain())
+                .subscribe(paths -> {
+                    if (paths == null) {
+                        moveComponent.path = Path.INVALID;
+                    } else if (paths.size() > 0) {
+                        moveComponent.path = paths.get(0);
+                    }
+                    actor.save(moveComponent);
+                }, throwable -> moveComponent.path = Path.INVALID);
 
-        Futures.addCallback(pathFuture, new FutureCallback<List<Path>>() {
-            @Override
-            public void onSuccess(List<Path> paths) {
-                if (paths == null) {
-                    moveComponent.path = Path.INVALID;
-                } else if (paths.size() > 0) {
-                    moveComponent.path = paths.get(0);
-                }
-                actor.save(moveComponent);
-            }
-
-            @Override
-            public void onFailure(Throwable t) {
-                moveComponent.path = Path.INVALID;
-            }
-        }, MoreExecutors.directExecutor());
```

Comment:
the current version of FindPathToNode no longer has a usage of an Executor like this.